x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -821,9 +823,36 @@ protected QueryTranslation asQuery(Range r, boolean onAggs) {
                 if (onAggs) {
                     aggFilter = new AggFilter(at.id().toString(), r.asScript());
                 } else {
+                    Holder<Object> lower = new Holder<>(valueOf(r.lower()));
```

Comment:
Why use the holder? Does the closure require a final variable?

Reply:
@costin yes.